win32ss/gdi/ntgdi/freetype.c

```diff
@@ -2994,38 +2988,91 @@ static unsigned int get_bezier_glyph_outline(FT_Outline *outline, unsigned int b
     return needed;
 }
 
-static INT
-IntRequestFontSize(PDC dc, FT_Face face, LONG Width, LONG Height)
+static FT_Error
+IntRequestFontSize(PDC dc, PFONTGDI FontGDI, LONG lfWidth, LONG lfHeight)
 {
+    FT_Error error;
     FT_Size_RequestRec  req;
+    FT_Face face = FontGDI->SharedFace->Face;
+    TT_OS2 *pOS2;
+    TT_HoriHeader *pHori;
+    FT_WinFNT_HeaderRec WinFNT;
+    LONG A, D, AD;
 
-    if (Width < 0)
-        Width = -Width;
+    if (lfWidth < 0)
+        lfWidth = -lfWidth;
 
-    if (Height < 0)
+    if (lfHeight == 0)
     {
-        Height = -Height;
+        lfHeight = dc->ppdev->devinfo.lfDefaultFont.lfHeight;
     }
-    if (Height == 0)
+    if (lfHeight == 0)
     {
-        Height = dc->ppdev->devinfo.lfDefaultFont.lfHeight;
+        lfHeight = lfWidth;
     }
-    if (Height == 0)
+
```

Comment:
The old code tried to limit the possible height and width to min: 1 and max: 0xFFFFU == USHORT_MAX (could be useful in case the user requests an "invalid" size, or the font file is corrupted and specifies wrong data). What do you think about that?

Reply:
Now tmEmHeight was bounded.